### PR TITLE
Bump dependencies to fix vulnerabilities

### PR DESCRIPTION
**Pull request by andrejatcastai**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -71,3 +71,8 @@ require (
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
+
+replace (
+	github.com/emicklei/go-restful/v3 => github.com/emicklei/go-restful/v3 v3.10.0
+	k8s.io/apimachinery => k8s.io/apimachinery v0.28.2
```
Comment (linkas45) [inline go.mod:77]: why do we need it in `replace` clause? Isn't it already picking up the correct version `v0.28.2` ?

Comment (andrejatcastai) [reply go.mod:77]: There are different versions involved due transitive dependencies. `replace` clause makes version usage consistent no matter what version comes transitively.

Comment (linkas45) [reply go.mod:77]: Then maybe it also make sense to do the same with `k8s.io/api` and` k8s.io/client-go`? So they all would be consistent and have the same version

Comment (andrejatcastai) [reply go.mod:77]: Added as suggested. However it makes even more sense to get rid of k8s client in spot-handler as it used only to fetching k8s version. The same can be achieved by using pure restful GET call to a k8s endpoint.

**Review by linkas45 — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (linkas45) [reply go.mod:77]: well, it's your call, I approve

**Pull request merged**